Assets/Plugins/LeapMotion/Modules/Hands/Scripts/HandBinder/Editor/HandBinderEditor.cs

```diff
@@ -0,0 +1,718 @@
+            globalFingerRotationOffset = serializedObject.FindProperty("GlobalFingerRotationOffset");
+            customBoneDefinitions = serializedObject.FindProperty("CustomBoneDefinitions");
+            handedness = serializedObject.FindProperty("handedness");
+            fineTuning = serializedObject.FindProperty("fineTuning");
+            debugOptions = serializedObject.FindProperty("debugOptions");
+            riggingOptions = serializedObject.FindProperty("riggingOptions");
+            armRigging = serializedObject.FindProperty("armRigging");
+            boundHand = serializedObject.FindProperty("boundHand");
+            offsets = serializedObject.FindProperty("offsets");
+
+            handTexture = Resources.Load<Texture>("Editor_hand");
+            buttonTexture = Resources.Load<Texture>("Editor_Documentation_Green_Upstate");
+            downstate = Resources.Load<Texture>("Editor_Documentation_Green_Downstate");
+            dividerLine = Resources.Load<Texture>("Editor_Divider_line");
+            subButton = Resources.Load<Texture>("secondary_button");
+        }
+
+        private void OnEnable() {
+            myTarget = (HandBinder)target;
+
+            if(myTarget.gameObject.name.ToUpper().Contains("Left".ToUpper())) {
+                myTarget.handedness = Chirality.Left;
+            }
+            if(myTarget.gameObject.name.ToUpper().Contains("Right".ToUpper())) {
+                myTarget.handedness = Chirality.Right;
+            }
+
+            serializedObject.Update();
+            SerializedProperties();
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        /// <summary>
+        /// Set the GUIStyle of the varius GUI elements we render
+        /// </summary>
+        private void StyleSetUp() {
+            buttonStyle = new GUIStyle(GUI.skin.button) {
+                alignment = TextAnchor.MiddleCenter,
+                normal = new GUIStyleState() {
+                    textColor = Color.black,
+                    background = (Texture2D)buttonTexture,
+                },
+                active = new GUIStyleState() {
+                    textColor = Color.white,
+                    background = (Texture2D)downstate,
+                },
+                fontSize = 20,
+                fontStyle = FontStyle.Bold,
+                stretchHeight = true,
+                fixedHeight = 50,
+            };
+
+            subButtonStyle = new GUIStyle(GUI.skin.button) {
+                alignment = TextAnchor.MiddleCenter,
+                normal = new GUIStyleState() {
+                    textColor = Color.white,
+                    background = (Texture2D)subButton,
+                },
+                stretchHeight = true,
+                fixedHeight = 20,
+            };
+
+            previousCol = GUI.color;
+        }
+
+        /// <summary>
+        /// Draw the inspector GUI
+        /// </summary>
+        public override void OnInspectorGUI() {
```

Comment:
There is an awful lot happening in this function. Maybe use your IDE refactoring tools to extract some methods as subfunctions or local functions so that you can give things names to explain what they draw (i.e. the code self-documents itself that way)?

Reply:
Great idea, I have split them into subfunctions as suggested.